lib/sequent/core/event_publisher.rb

```diff
@@ -0,0 +1,44 @@
+module Sequent
+  module Core
+    class EventPublisher
+      class PublishEventError < RuntimeError
+        attr_reader :event_handler_class, :event
+
+        def initialize(event_handler_class, event)
+          @event_handler_class = event_handler_class
+          @event = event
+        end
+
+        def message
+          "Event Handler: #{@event_handler_class.inspect}\nEvent: #{@event.inspect}\nCause: #{cause.inspect}"
+        end
+      end
+
+      def initialize
+        @events_queue = Queue.new
+      end
+      def configuration
+        Sequent.configuration
+      end
+
+      def publish_events(events)
+        return if configuration.disable_event_handlers
+        events.each { |event| @events_queue.push(event) }
+        process_events
+      end
+
+      def process_events
+        while(!@events_queue.empty?) do
+          event = @events_queue.pop
+          configuration.event_handlers.each do |handler|
+            begin
+              handler.handle_message event
+            rescue
```

Comment:
This rescue can be moved down, this makes the code a bit more readable

```
def process_events
rescue
end
```

Reply:
Dat kan niet: dan hebben we geen referentie naar `event` meer.

Reply:
Doh, you are right